db/kv_dense_block.go

```diff
@@ -264,6 +262,42 @@ func (d *DenseBlock) RemoveEntrySet(entries []*LogEntry, bucket base.Bucket) (pe
 	return pendingRemoval, nil
 }
 
+// Removes all entries greater than vb, seq from the block.  Returns rollbackComplete=true if it finds a seq
+// in the block for the vbucket where seq <= rollbackSeq
+func (d *DenseBlock) RollbackTo(rollbackVbNo uint16, rollbackSeq uint64, bucket base.Bucket) (rollbackComplete bool, err error) {
+
+	numRemoved := 0
+	numRemoved, rollbackComplete = d.rollbackEntries(rollbackVbNo, rollbackSeq)
+
+	// If nothing was removed, don't update the block
+	if numRemoved == 0 {
+		return rollbackComplete, nil
+	}
+
+	casOut, writeErr := base.WriteCasRaw(bucket, d.Key, d.value, d.cas, 0, func(value []byte) (updatedValue []byte, err error) {
+		// Note: The following is invoked upon cas failure - may be called multiple times
```

Comment:
this might be more readable as:

```
casFunc := func(value []byte) (updatedValue []byte, err error) {
        ...
}

 base.WriteCasRaw(
    bucket, 
    d.Key, 
    d.value, 
    d.cas, 
    0, 
    casFunc,
)
```